saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/OpenSamlAuthenticationRequestFactory.java

```diff
@@ -16,16 +16,32 @@
 
 package org.springframework.security.saml2.provider.service.authentication;
 
-import org.opensaml.saml.common.xml.SAMLConstants;
-import org.springframework.util.Assert;
-
+import org.apache.commons.codec.binary.Base64;
```

Comment:
Is there a need to use commons-codec here?

Reply:
we are finding that `java.util.Base64` doesn't encompass all of the providers as there are minor differences in encoding.

The Commons Codec is used by OpenSAML underneath. I added a test case:
https://github.com/spring-projects/spring-security/blob/9d26f12e86d0349af33cea3b2de14bb491b16186/saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2UtilsTests.java#L65-L67

We can investigate further, but there is a risk of regression using a different Base64 than OpenSAML

Reply:
I've swapped this instance for `java.util.Base64`